flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/source/reader/deserializer/AvroToRowDataConverters.java

```diff
@@ -0,0 +1,406 @@
+            tempMicros = (Long) object;
+            if (precision == millisPrecision) {
+                // If millisecond precision.
+                return TimestampData.fromEpochMillis(tempMicros);
+            }
+        } else if (object instanceof Instant) {
+            // Precision is automatically transferred.
+            return TimestampData.fromInstant(((Instant) object));
+        } else if (object instanceof String || object instanceof Utf8) {
+            // ---- DATETIME TYPE Conversion ----
+            // LocalDateTime.parse is also responsible for validating the string passed.
+            // If the text cannot be parsed DateTimeParseException is thrown.
+            // Formatting,
+            // according to
+            // https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#datetime_type.
+            String tsValue;
+            if (object instanceof String) {
+                tsValue = (String) object;
+            } else {
+                tsValue = ((Utf8) object).toString();
+            }
+            try {
+                // LocalDateTime will handle the precision.
+                return TimestampData.fromLocalDateTime(
+                        LocalDateTime.parse(
+                                tsValue,
+                                DateTimeFormatter.ofPattern(
+                                        "yyyy-M[M]-d[d][[' ']['T']['t']H[H]':'m[m]':'s[s]['.'SSSSSS]['.'SSSSS]['.'SSSS]['.'SSS]['.'SS]['.'S]]")));
+            } catch (DateTimeParseException e) {
+                String invalidFormatError =
+                        String.format(
+                                "The datetime string obtained %s, is of invalid format.", object);
+                // https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#datetime_type for more details.
+                String supportedFormat =
+                        "civil_date_part[time_part]\n"
+                                + "civil_date_part: YYYY-[M]M-[D]D\n"
+                                + "time_part: { |T|t}[H]H:[M]M:[S]S[.F]\n"
+                                + "YYYY: Four-digit year.\n"
+                                + "[M]M: One or two digit month.\n"
+                                + "[D]D: One or two digit day.\n"
+                                + "{ |T|t}: A space or a T or t separator. The T and t separators are flags for time.\n"
+                                + "[H]H: One or two digit hour (valid values from 00 to 23).\n"
+                                + "[M]M: One or two digit minutes (valid values from 00 to 59).\n"
+                                + "[S]S: One or two digit seconds (valid values from 00 to 60).\n"
+                                + "[.F]: Up to six fractional digits (microsecond precision).";
+                LOG.error(
+                        String.format(
+                                "%s%nSupported Format:%n%s", invalidFormatError, supportedFormat));
```

Comment:
Why log error here but not elsewhere? Lets be consistent if possible. If there isnt any additional info to be shared beyond the error message then you can skip error log.
Also, like previously mentioned, this log or exception do not tell where/when did the issue arise. Lets add more context.

Reply:
LOG here is more detailed than others (taken from BQ data type implementation). Would not be possible to include it as a part of error message. Can add such detailed logs for other types as well if needed.